flusher.go

```diff
@@ -15,23 +16,32 @@ import (
 
 	"github.com/Sirupsen/logrus"
 	"github.com/stripe/veneur/samplers"
+	"github.com/stripe/veneur/ssf"
+	"github.com/stripe/veneur/trace"
 )
 
 // Flush takes the slices of metrics, combines then and marshals them to json
 // for posting to Datadog.
 func (s *Server) Flush(interval time.Duration, metricLimit int) {
+	trace := trace.StartTrace("flush")
+	defer trace.Record("veneur.flush.trace", []*ssf.SSFTag{})
+
 	// right now we have only one destination plugin
 	// but eventually, this is where we would loop over our supported
 	// destinations
 	if s.IsLocal() {
-		s.FlushLocal(interval, metricLimit)
+		s.FlushLocal(trace.Attach(context.Background()), interval, metricLimit)
 	} else {
-		s.FlushGlobal(interval, metricLimit)
+		s.FlushGlobal(trace.Attach(context.Background()), interval, metricLimit)
 	}
 }
 
-func (s *Server) FlushGlobal(interval time.Duration, metricLimit int) {
+func (s *Server) FlushGlobal(ctx context.Context, interval time.Duration, metricLimit int) {
+	trace := trace.SpanFromContext(ctx)
```

Comment:
It's _probably not worth doing since we'll eventually use OpenTracing, but is there ever a situation where you wouldn't immediately `defer trace.Record` such that we could combine them?

(Again, probably moot.)